frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -226,8 +254,40 @@ const edgeDragSourceSpec = (
 
 const createConnectorCallback = (serviceBinding: boolean) => (
   source: Node,
-  target: Node,
+  target: Node | Graph,
+  dragEvent,
+  choice,
 ): any[] | null => {
+  const onKebabOptionClick = (option: KebabOption, sourceNode: Node) => {
+    if (option.callback) {
+      option.callback();
+    }
+    if (option.href) {
+      const sourceObj = _.get(sourceNode.getData(), ['resources', 'obj'], null);
+      const hrefWithContext = `${option.href}&contextSource=${referenceFor(sourceObj)}/${
+        sourceObj.metadata.name
+      }isKnativeDisabled=true`;
+      history.push(hrefWithContext);
+    }
+  };
+
+  if (isGraph(target)) {
+    if (choice) {
+      onKebabOptionClick(choice, source);
+    }
+    return graphActions(target.getController().getElements());
+  }
+  if (target.isGroup()) {
+    const applicationGroup: TopologyApplicationObject = {
+      id: target.getId(),
+      name: target.getLabel(),
+      resources: target.getNodes().map((currNode) => currNode.getData()),
+    };
```

Comment:
Why is this here? i see it's copied code. @jeff-phillips-18 why do we need this? Shouldn't we be using the node data directly?

Reply:
we could use node data directly but that would require modifying the groupContextMenu and groupActions files. so I created this object here.
I've made the necessary modifications to the functions and groupContextMenu is directly used now.